packages/wasm-dpp/src/identity/state_transition/asset_lock_proof/instant/validation.rs

```diff
@@ -0,0 +1,46 @@
+use dpp::identity::state_transition::asset_lock_proof::{
+    AssetLockTransactionValidator, InstantAssetLockProofStructureValidator,
+};
+use dpp::state_transition::state_transition_execution_context::StateTransitionExecutionContext;
+use dpp::ProtocolError;
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+
+use crate::buffer::Buffer;
+use crate::utils::ToSerdeJSONExt;
+use crate::{
+    errors::from_dpp_err,
+    state_repository::{ExternalStateRepositoryLike, ExternalStateRepositoryLikeWrapper},
+    validation::ValidationResultWasm,
+    StateTransitionExecutionContextWasm,
+};
+
+#[wasm_bindgen(js_name=validateInstantAssetLockProofStructure)]
+pub async fn validate_instant_asset_lock_proof_structure(
```

Comment:
Same as above

Reply:
Fixed. But please check `identity_create_transition_basic_validator`. 
I had to remove `Arc<>` from `ProtocolVersionValidator` because I needed to mutate `current_protocol_version` variable in `validate` function, and `Arc` disallows that.

Not sure if this is acceptable though